gfw_pixetl/decorators.py

```diff
@@ -43,19 +49,33 @@ def process_func(q, *args, **kwargs):
         q.put((ret, error))
 
     # register original function with different name
-    # in sys.modules so it is pickable
+    # in sys.modules so it is picklable
     process_func.__name__ = func.__name__ + "processify_func"
     setattr(sys.modules[__name__], process_func.__name__, process_func)
 
     @wraps(func)
     def wrapper(*args, **kwargs):
         q = Queue()
         p = Process(target=process_func, args=[q] + list(args), kwargs=kwargs)
+
+        error = None
+        ret = None
+        untimely_death = False
+
         p.start()
-        ret, error = q.get()
-        p.join()
 
-        if error:
+        while p.is_alive():
+            p.join(timeout=60)  # TODO: Make configurable
```

Comment:
what happens if a process takes longer than 60 sec? we use this for the transform stage which often takes longer.

Reply:
If it's still running at 60s, p.join will return, p.exitcode will be None, and it will loop. So we're seeing if the process has ended (one way or another) every 60s.

Reply:
Yes, it's just a Queue object we pass in to the function as an argument. Though after scrutinizing this a bunch I realize that in the improbable case that the exception-handling code (the calls to sys.exc_info() and traceback.format_tb()) itself throws an exception, the call to q.get() will block forever (because q.put() will never be called). Maybe I'm being paranoid? I'll add a check for exit_code > 0 just in case.

Reply:
Actually I'll just add a timeout to q.get().